### PR TITLE
Text To Speech

### PR DESCRIPTION
**Pull request by ririchiyo**
dectalk.

### CODE REVIEW

**Comment by SysadminJeroen** [comment]
So, what is DecTalk?

**Comment by PJB3005** [comment]
Text to speech.

**Comment by Konater** [comment]
AEIOU

**Comment by SysadminJeroen** [comment]
So, it is a way to play TTS ingame? 

Don't we have the AI's TTS for that?

**Comment by clusterfack** [comment]
AI doesn't have text 2 speech it has premade vox

**Comment by SysadminJeroen** [comment]
@clusterfack With someone working on it, it can be made into TTS.

**Comment by clusterfack** [comment]
Yeah why would you bother when there are better TTS out there already?

**Comment by Iamgoofball** [comment]
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA

**Comment by Iamgoofball** [comment]
@clusterfack dectalk is THE spaceman TTS

**Comment by clusterfack** [comment]
@iamgoofball I know, that's why I suggested riri to add it

File: code/modules/admin/admin_verbs.dm [hunk 1]
```diff
@@ -1212,3 +1213,34 @@ var/list/admin_verbs_mod = list(
 	to_chat(src, "Attempting to load [AM.name] ([AM.file_path])...")
 	createRandomZlevel(override, AM, usr)
 	to_chat(src, "The away mission has been generated on z-level [world.maxz] [AM.location ? "([formatJumpTo(AM.location)])" : ""]")
+
+/client/proc/cmd_dectalk()
```
Comment (ComicIronic) [inline code/modules/admin/admin_verbs.dm]: Check holder for this.

**Comment by ComicIronic** [comment]
There are many, many good reasons not to polish a turd. What's the speed like for this?

**Comment by Iamgoofball** [comment]
looks like it offloads it all onto a TTS server

**Comment by ComicIronic** [comment]
I know that, I was asking about how fast that was i.e. the roundtrip.

**Comment by 9600bauds** [comment]
What's it sound like

**Comment by SysadminJeroen** [comment]
@9600bauds Google: "Moonbase Alpha text to speech"

File: tools/DecTalk_Server/DecTalk_Server/DecTalk_Server/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
+﻿using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+// General Information about an assembly is controlled through the following 
+// set of attributes. Change these attribute values to modify the information
+// associated with an assembly.
+[assembly: AssemblyTitle("ConsoleApplication1")]
```
Comment (ComicIronic) [inline tools/DecTalk_Server/DecTalk_Server/DecTalk_Server/Properties/AssemblyInfo.cs]: Fix pls

**Comment by NigglyWiggly** [comment]
Jesus christ please no

File: code/modules/admin/admin_verbs.dm (hunk 1 above)
Comment (PJB3005) [inline code/modules/admin/admin_verbs.dm]: No, `check_rights()`

**Comment by clusterfack** [comment]
Niggy "no fun allowed" Wiggly has arrived on the station.

**Comment by NigglyWiggly** [comment]
Don't get me wrong.
I love dectalk.
It's fucking hilarious.
But please jesus do not make it a thing in spess, at least if you do, make it a button or something.
;###################################################################
over and over and over is only funny for so long.

**Comment by clusterfack** [comment]
Obviously if it was ever implemented beyond admins it would have a preference toggle

**Comment by 9600bauds** [comment]
can't be worse than "Nyan~_Neko_Sugar_Girls_-Episode_2-.ogg" being played serverwide

**Comment by ihadtoregisterforthis** [comment]
Why not start by giving it to the AI?

**Comment by Imagebard** [comment]
I looked through my entire reactions image list and none of them can list how uneasy I feel about this

**Comment by ComicIronic** [comment]
A S S B L A S T U S A

**Comment by AutismSpider** [comment]
Boy i cant wait to spam woody got wood over and over with this.

show me the butts

**Comment by The-Albinobigfoot** [comment]
> admin only

**Comment by clusterfack** [comment]
If it is added to player speech we will deinitely add a cancel button next to everyone's messages in case they are too long or obnoxious.

**Comment by SkowronX** [comment]
what

**Comment by TheEliminator16** [comment]
Oh jesus christ
HERE COMES ANOTHER CHINESE EARTHQUAKE
EBRBRBRBRBRBRBRBRBRBRBRBRBRBRBRBRBRBRBRBR

**Comment by Probe1** [comment]
:+1:  if it works without fucking anything up

It'd be a great thing for AIs

**Comment by ghost** [comment]
John Madden John Madden John Madden

**Comment by ScrambledSteak** [comment]
oh my god
can we have the spess opera now? This is the moonbase alpha tts, right?

**Comment by clusterfack** [comment]
Yes

**Comment by Iamgoofball** [comment]
Guys this isnt dectalk its fonixtalk

**Comment by clusterfack** [comment]
dectalk is owned by fonixtalk

**Comment by FudgePucker** [comment]
John Madden John Madden

**Comment by clusterfack** [comment]
We probably wont end up using dectalk exactly due to copyright claims, but we have plans for another variant of tts that we can try out where you can install various voices.

Long term I want to be able to do things like apply a pitch and voice that players can configure for themselves, and possibly apply race filters for speech.

**Comment by SkowronX** [comment]
Still don't get the purpose of this feature.

**Comment by Imagebard** [comment]
Hatred for people

**Comment by clusterfack** [comment]
How do you not get the purpose of it?

**Comment by Imagebard** [comment]
**Hatred for your own playerbase**

**Comment by clusterfack** [comment]
You have all only ever seen the very tip of the full force of my malicious intent

**Comment by Imagebard** [comment]
Because we don't allow you to unleash all of it.

**Comment by Iamgoofball** [comment]
@clusterfack Fonixtalk isn't nearly as powerful as real dectalk.

**Comment by clusterfack** [comment]
@Iamgoofball Well as I said it doesn't really matter at the current moment since this is being reformulated anyway

**Comment by Iamgoofball** [comment]
reformulate this:

8=====D O: <- goofball

**Comment by clusterfack** [comment]
Okay, reformulated

**Comment by Iamgoofball** [comment]
thanks

**Comment by NigglyWiggly** [comment]
What have you done

**Comment by NigglyWiggly** [comment]
If only it worked

**Comment by clusterfack** [comment]
Where did I say you would be able to use it immediately after I merged it?

**Comment by NigglyWiggly** [comment]
That's typically how almost every single thing in spess works

**Comment by clusterfack** [comment]
Yeah everything made in fucking byond code maybe which this OBVIOUSLY ISN'T.

**Comment by gbasood** [comment]
~~doesn't this still not work~~
~~because I tried it and nothing happened~~
nvm it just doesnt work on single syllables

**Comment by gbasood** [comment]
although dectalk special arguments like [:dv sx 0] do nothing

**Comment by clusterfack** [comment]
That's because it's not actually dectalk

**Comment by gbasood** [comment]
rip

**Comment by StarSmasher44** [comment]
So this is nonfunctional and will continue to be that way? Otherwise, how would I go around making this work? I can't find anything about setting up said Dectalk thing

**Comment by ComicIronic** [comment]
Compile and run the C# solution, and then use the dectalk verb from inside SS13. Be sure to set up the url correctly, including the port. There's no incode support currently for actually speaking with it.

**Pull request merged**